test/python/circuit/test_controlled_gate.py

```diff
@@ -1772,121 +1798,46 @@ def test_control_label_1(self, gate, args):
-        self.assertIsInstance(annotated, AnnotatedOperation)
-        self.assertEqual(Operator(controlled), Operator(annotated))
-
-    def test_controlled_rz(self):
-        """Test the creation of a controlled RZ gate."""
-        theta = 0.5
-        controlled = RZGate(theta).control(annotated=False)
-        annotated = RZGate(theta).control(annotated=True)
-        self.assertNotIsInstance(controlled, AnnotatedOperation)
-        self.assertIsInstance(annotated, AnnotatedOperation)
-        self.assertEqual(Operator(controlled), Operator(annotated))
-
-    def test_controlled_ry(self):
-        """Test the creation of a controlled RY gate."""
-        theta = 0.5
-        controlled = RYGate(theta).control(annotated=False)
-        annotated = RYGate(theta).control(annotated=True)
-        self.assertNotIsInstance(controlled, AnnotatedOperation)
-        self.assertIsInstance(annotated, AnnotatedOperation)
-        self.assertEqual(Operator(controlled), Operator(annotated))
-
-    def test_controlled_rx(self):
-        """Test the creation of a controlled RX gate."""
-        theta = 0.5
-        controlled = RXGate(theta).control(annotated=False)
-        annotated = RXGate(theta).control(annotated=True)
-        self.assertNotIsInstance(controlled, AnnotatedOperation)
-        self.assertIsInstance(annotated, AnnotatedOperation)
-        self.assertEqual(Operator(controlled), Operator(annotated))
-
-    def test_controlled_u(self):
-        """Test the creation of a controlled U gate."""
-        theta, phi, lamb = 0.1, 0.2, 0.3
-        controlled = UGate(theta, phi, lamb).control(annotated=False)
-        annotated = UGate(theta, phi, lamb).control(annotated=True)
-        self.assertNotIsInstance(controlled, AnnotatedOperation)
-        self.assertIsInstance(annotated, AnnotatedOperation)
-        self.assertEqual(Operator(controlled), Operator(annotated))
-
-    def test_controlled_u3(self):
-        """Test the creation of a controlled U3 gate."""
-        theta, phi, lamb = 0.1, 0.2, 0.3
-        controlled = U3Gate(theta, phi, lamb).control(annotated=False)
-        annotated = U3Gate(theta, phi, lamb).control(annotated=True)
-        self.assertNotIsInstance(controlled, AnnotatedOperation)
-        self.assertIsInstance(annotated, AnnotatedOperation)
-        self.assertEqual(Operator(controlled), Operator(annotated))
-
-    def test_controlled_cx(self):
-        """Test creation of controlled cx gate"""
-        controlled = CXGate().control(annotated=False)
-        annotated = CXGate().control(annotated=True)
+    @data(
+        (XGate(), 1),
+        (XGate(), 2),
+        (XGate(), 4),
+        (PhaseGate(0.5), 1),
+        (PhaseGate(0.5), 2),
+        (U1Gate(0.5), 1),
+        (U1Gate(0.5), 2),
+        (UGate(0.1, 0.2, 0.3), 1),
+        (CXGate(), 1),
+        (CXGate(), 2),
+    )
+    @unpack
+    def test_native_controlled_gates(self, base_gate, num_controls):
```

Comment:
these tests are for gates that have multi-controlled version? perhaps it's better to say "multi".
also, I think you should add here:
CCX, C3X, C4X